pyiode/objects/equation.pyx

```diff
@@ -122,6 +125,8 @@ cdef class Equation:
     def __dealloc__(self):
         del self.c_equation
 
+    # GDM> what is the point of the name argument if you cannot change the name (or maybe I misunderstand the doctest)?
```

Comment:
Indeed, the argument *name* is obsolete since I pushed the PR #566. I will remove the argument *name*.

@jmpplan will correct me if necessary (see below).
FYI, even the name (= its endogeneous variable) cannot be changed, the C struct `EQ` (defined file [api/iode.h](https://github.com/plan-be/iode/blob/master/api/iode.h#L891)) does NOT contain an attribute *name* or *endo*.  This is historical since @jmpplan wanted first to consider the possibility to have several endogeneous variables in the same equation (if I remember correctly). But since this was way too complicated to handle, it was finally decided to freeze the number of endogeneous to one and to prevent users to rename an equation in the global Equations workspace.
The current Python set_lec() method calls a C function requiring to pass the *endo* (= name of the equation) because it is NOT stored in the C `EQ` struct.
But now I have a *endogenous* attribute in my Python Equation class, the argument *name* is no longer required.

Reply:
Change made in PR #580